### PR TITLE
[feat]: [PL-27423]: OpenAPI Implementation for Resource Groups through Server Stub

### PR DESCRIPTION
**Pull request by mankrit-singh**
## Describe your changes
Implemented Resource Groups API through server-stub generation.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37108)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 20 Issues 
* Code issue 19 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37108)

File: platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v3/api/OrgResourceGroupsApiImpl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    ResourceGroupsResponse resourceGroupsResponse = ResourceGroupApiUtils.getResourceGroupResponse(
+        resourceGroupService.create(resourceGroupRequest.getResourceGroup(), false));
+    return Response.status(201).entity(resourceGroupsResponse).build();
+  }
+
+  @Override
+  @NGAccessControlCheck(resourceType = RESOURCE_GROUP, permission = DELETE_RESOURCEGROUP_PERMISSION)
+  public Response deleteResourceGroupOrg(String org, String resourceGroup, String account) {
+    ResourceGroupsResponse resourceGroupsResponse = ResourceGroupApiUtils.getResourceGroupResponse(
+        resourceGroupService.get(Scope.of(account, null, null), resourceGroup, NO_FILTER).orElse(null));
```
Comment (ashishsanodia-harness) [inline platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v3/api/OrgResourceGroupsApiImpl.java:66]: org should not be null here

Comment (ashishsanodia-harness) [reply platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v3/api/OrgResourceGroupsApiImpl.java:66]: check other places as well. add unit test for the same

File: platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v3/api/ProjectResourceGroupsApiImpl.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ResourceGroupsResponse resourceGroupsResponse = ResourceGroupApiUtils.getResourceGroupResponse(
+        resourceGroupService.create(resourceGroupRequest.getResourceGroup(), false));
+    return Response.status(201).entity(resourceGroupsResponse).build();
+  }
+
+  @Override
+  @NGAccessControlCheck(resourceType = RESOURCE_GROUP, permission = DELETE_RESOURCEGROUP_PERMISSION)
+  public Response deleteResourceGroupProject(String org, String project, String resourceGroup, String account) {
+    ResourceGroupsResponse resourceGroupsResponse = ResourceGroupApiUtils.getResourceGroupResponse(
+        resourceGroupService.get(Scope.of(account, null, null), resourceGroup, NO_FILTER).orElse(null));
```
Comment (ashishsanodia-harness) [inline platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v3/api/ProjectResourceGroupsApiImpl.java:67]: here as well org and project both should not be null.

**Review by reetika04 — APPROVED** [review]
> Reviewed 1 of 10 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37108)* status: 10 file(s) to review. 1 of 2 approvals obtained.

**Review by ashishsanodia-harness — APPROVED** [review]
> Reviewed 1 of 10 files at r1, 9 of 9 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37108)* status: :shipit: complete! ✓

**Pull request merged**